client/src/components/HistoryViewer/HistoryViewerToolbar.js

```diff
@@ -0,0 +1,98 @@
+import React, { Component, PropTypes } from 'react';
+import { compose } from 'redux';
+import { connect } from 'react-redux';
+import i18n from 'i18n';
+import { inject } from 'lib/Injector';
+import { addMessage, setCurrentPage, setCurrentVersion } from 'state/historyviewer/HistoryViewerActions';
+
+class HistoryViewerToolbar extends Component {
+  constructor(props) {
+    super(props);
+
+    this.handleRevert = this.handleRevert.bind(this);
+  }
+
+  /**
+   * Triggers a revert action to be performed for the current record's version
+   */
+  handleRevert() {
+    const { actions: { revertToVersion }, onAfterRevert, recordId, versionId } = this.props;
+
+    revertToVersion(recordId, versionId, 'DRAFT', 'DRAFT');
+
+    onAfterRevert(versionId);
+  }
```

Comment:
`revertToVersion` returns a Promise, right? And the mutation returns `ID`. So couldn't this be:

```js
revertToVersion(recordId, versionId, 'DRAFT', 'DRAFT')
  .then(onAfterRevert);
```

Reply:
Tried changing it, but it breaks the test.

Reply:
That would be my expectation, because you've changed it from an asynchronous call to a synchronous one. A failing test shouldn't be a reason to not implement it. It should be a reason to update the test. :) 

At the very least, change the prop name from `onAfterRevert` to something else, because it's if it's being invoked asynchronously, that's not an honest name.

Reply:
Actually, `onRevert` is a more appropriate name.